src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java

```diff
@@ -156,24 +159,21 @@ private static boolean addToAvailableGames(
    */
   public GameData getGameData(final String gameName) {
     return Optional.ofNullable(availableGames.get(gameName))
-        .map(AvailableGames::getGameDataFromXml)
+        .map(uri -> parse(uri).orElse(null))
```

Comment:
```java
Optional.ofNullable(availableGames.get(gameName))
  .map(AvailableGames::parse)
  .orElse(Optional.empty)
  .orElse(null)
```
? 😅 
It would be great if we could avoid returning null in general, and simply return an optional